### ISSUE TITLE
Confusion between R parallel and R orthogonal in section 10.2

### ISSUE THREAD

**Issue by IceTDrinker** [RayTracing/raytracing.github.io#526]
Hello,

I re-did the math (as proposed in the 10.2 section on refraction) and there seems to be a confusion on what R parallel and R orthogonal are.

The book defines R parallel as being parallel to the normal.

But from the book when we look at the expression in terms of known quantities we have :
> We can now rewrite R′∥ in terms of known quantities:
>
**R′** ∥=η/η′(**R**+(−**R**⋅**n**)**n**)

**R**.**n** is the component of **R** along **n** and we subtract that from **R** meaning that the right handside of the above equation is tangential (or along the surface) which contradicts the definition of R parallel being parallel to the normal.

I can't recall from past physics lecture if the convention is to consider direction with respect to the surface (which would make sense here as R parallel is parallel to the surface/tangent and R orthogonal is orthogonal to the surface)

Let me know if that makes sense or if I missed something.

Cheers

Edit : typo

**Comment by hollasch** [comment]
The associated figure could also be improved, as R and R' are both shown as bidirectional, but the direction of the rays is important to understand the calculations. Is R pointing to the surface or away from it? I imagine that R' is the ray pointing into the refracting medium (downwards), but this should also be made explicit in the diagram.

**Comment by IceTDrinker** [comment]
Yes you are correct R and R' are not drawn and it could be complicated for people who have never seen wave propagation.

Maybe there was something about Fermat's principle in there ?

**Comment by trevordblack** [comment]
I skipped a step because the math was already really heavy

`R′ ∥=η/η′(R+(R⋅n')n)`

where `n' = -n`

`R′ ∥=η/η′(R+(−R⋅n)n)`

**Comment by trevordblack** [comment]
`R` and `N` point in opposing directions,
so `R⋅N` < 0, hence
`-R⋅N` > 0

The diagram could have R and R' with correct directions, but refraction is _hard_. And we should be careful about where what we change

**Comment by IceTDrinker** [comment]
It does not change the fact that R parallel is not parallel to **n** :)

**Comment by IceTDrinker** [comment]
And I think it yields the same result as long as dot(R, R’) > 0

**Comment by micgdev** [comment]
I agree with this:

> the convention is to consider direction with respect to the surface (which would make sense here as R parallel is parallel to the surface/tangent and R orthogonal is orthogonal to the surface)

Here is the other side of the problem:
It is written in the book that <b>R'</b><sub>⊥</sub>=−sqrt(1−|<b>R′</b><sub>∥</sub>|<sup>2</sup>)<b>n</b>.
However, sqrt(1−|<b>R′</b><sub>∥</sub>|<sup>2</sup>) is a scalar, so <b>R'</b><sub>⊥</sub> must be parallel to **n**, i.e. parallel to <b>n'</b>.

Combined with @IceTDrinker's argument,

> R′ ∥=η/η′(R+(−R⋅n)n)
R.n is the component of R along n and we subtract that from R meaning that the right handside of the above equation is tangential (or along the surface) which contradicts the definition of R parallel being parallel to the normal.

, there is clearly a confusion of convention here.

**Comment by trevordblack** [comment]
Sorry Folks. I was totally wrong on this one.

You're all correct.

When I worked out the proof to confirm functionality, I used R_perp to represent perpendicular with the surface, and r_parallel to represent parallel to the surface.

After proving it, I realized that it would be more intuitive for the reader to work from the (already available) local normal.

I switched all of the math over, but must have forget to update the subscript.

So, that's pretty embarrassing.

Thanks everyone for pointing this out so others aren't taught incorrectly.
Fix is coming in the new minor release coming _sometime_ this week or next.
